### PR TITLE
Split the Ssrmatching module between code and grammar rules.

### LINKED ISSUE #7857: Split ssrmatching.ml4
This file is a bit of a pain as merlin breaks on the syntax extensions, so when changing an API (eg deprecate something) I have to recompile once/error to find each error.
I tried splitting it myself but the interlacing with syntax extension stuff was too much for me.

A way to get merlin to work in ml4 files would also work but I'm not hopeful about it.

### PR DESCRIPTION
**Pull request by ppedrot**
Fixes #7857.

### CODE REVIEW

File: plugins/ssr/ssrparser.ml4 [hunk 1]
```diff
@@ -2050,7 +2051,7 @@ END
 (** Rewrite arguments *)
 
 let pr_option f = function None -> mt() | Some x -> f x
-let pr_pattern_squarep= pr_option (fun r -> str "[" ++ pr_rpattern r ++ str "]")
+let pr_pattern_squarep= pr_option (fun r -> str "[" ++ Ssrmatching_plugin.Ssrmatching.pr_rpattern r ++ str "]")
```
Comment (gares) [inline plugins/ssr/ssrparser.ml4]: There is an open up there, I guess this is line is spurious

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
CI looks good, I've restarted a "edge flambda whatever" job, all the rest was green

File: plugins/ssr/ssrparser.ml4 (hunk 1 above)
Comment (ppedrot) [reply plugins/ssr/ssrparser.ml4]: Do you want me to fix this?

**Comment by ppedrot** [comment]
Nitpick addressed, ready to go.

**Comment by gares** [comment]
thanks

**Pull request merged**